7_react/react1/week2/src/components/TodoList.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { Component } from "react";
+import "./TodoList.css";
+
+const todos = [
+  {
+    id: 1,
+    description: "Attend family meeting",
+    deadline: "Sun Mar 10 2024",
+    done: false,
+  },
+  {
+    id: 2,
+    description: "Get some groceries",
+    deadline: "Mon Mar 11 2024",
+    done: false,
+  },
+  {
+    id: 3,
+    description: "Walk my dog",
+    deadline: "Tue Mar 12 2024",
+    done: false,
+  },
+];
+
+class TodoList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: todos,
+      timer: 0,
+      description: "",
+      deadline: "",
+    };
+  }
+
+  componentDidMount() {
+    this.timerID = setInterval(() => this.tick(), 1000);
```

Comment:
Well done on implementing the timer 👏! You are the first so far. It's interesting that you are using `class components` Is there a specific reason for this choice? No worries, it's still correct. However, it's worth noting that `class components` are not as commonly used for this purpose nowadays. Here's a more modern approach using `functional components` and `useEffect`. I highly recommend giving it a look when you have time :) https://legacy.reactjs.org/docs/hooks-effect.html.

Reply:
okay, I will definitely take it into consideration.